examples/model_compress/amc/amc_search.py

```diff
@@ -28,27 +29,29 @@ def parse_args():
     parser.add_argument('--train_episode', default=800, type=int, help='number of training episode')
     parser.add_argument('--n_gpu', default=1, type=int, help='number of gpu to use')
     parser.add_argument('--n_worker', default=16, type=int, help='number of data loader worker')
-    parser.add_argument('--job', default='train_export', type=str, choices=['train_export', 'export_only'],
-                        help='search best pruning policy and export or just export model with searched policy')
-    parser.add_argument('--export_path', default=None, type=str, help='path for exporting models')
-    parser.add_argument('--searched_model_path', default=None, type=str, help='path for searched best wrapped model')
+    parser.add_argument('--suffix', default=None, type=str, help='suffix of auto-generated log directory')
 
     return parser.parse_args()
 
 
 def get_model_and_checkpoint(model, dataset, checkpoint_path, n_gpu=1):
-    if model == 'mobilenet' and dataset == 'imagenet':
-        from mobilenet import MobileNet
-        net = MobileNet(n_class=1000)
-    elif model == 'mobilenetv2' and dataset == 'imagenet':
-        from mobilenet_v2 import MobileNetV2
-        net = MobileNetV2(n_class=1000)
-    elif model == 'mobilenet' and dataset == 'cifar10':
+    if dataset == 'imagenet':
+        n_class = 1000
+    elif dataset == 'cifar10':
+        n_class = 10
+    else:
+        raise ValueError('unsupported dataset')
+
+    if model == 'mobilenet':
         from mobilenet import MobileNet
-        net = MobileNet(n_class=10)
-    elif model == 'mobilenetv2' and dataset == 'cifar10':
+        net = MobileNet(n_class=n_class)
+    elif model == 'mobilenetv2':
         from mobilenet_v2 import MobileNetV2
-        net = MobileNetV2(n_class=10)
+        net = MobileNetV2(n_class=n_class)
+    elif model.startswith('resnet'):
+        net = resnet.__dict__[model](pretrained=True)
+        in_features = net.fc.in_features
+        net.fc = nn.Linear(in_features, n_class)
```

Comment:
why the `resnet` models need to modify the last linear layer in such a way and the `mobilenet` models don't ？Can they change the number of the classes in the same way?

Reply:
the mobilenet code is copied from amc repo, it does not load imagenet pretrained weight because the repo provides an amc pretrained weight.
for torchvision resnet implementation, if we want to load pretrained weight, we need to set num_classes = 1000, then replace  fc after loading pretrained weight.  Loading pretrained weight should be helpful.

Reply:
Got it!